File: dotnet/BeakPeekApi/Models/Pentad.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace BeakPeekApi.Models
{
    [Index(nameof(Pentad_Allocation), IsUnique = true)]
    public class Pentad
    {
        [Key]
        public required string Pentad_Allocation { get; set; }

        // Number before the underscore
        public int Pentad_Longitude { get; set; }

        // Number after the underscore
        public int Pentad_Latitude { get; set; }

        public required ProvinceList Province { get; set; }

        public int Total_Cards { get; set; }

        public PentadDto ToDto()
        {
            return new PentadDto
            {
                Pentad_Allocation = this.Pentad_Allocation,
                Pentad_Longitude = this.Pentad_Longitude,
                Pentad_Latitude = this.Pentad_Latitude,
                Province = this.Province.ToDto(),
                Total_Cards = this.Total_Cards
            };
        }

    }


}

File: beakpeek/lib/Model/BirdInfo/pentad.dart
import 'package:beakpeek/Model/BirdInfo/province.dart';

class Pentad {
  Pentad({
    required this.pentadAllocation,
    required this.pentadLongitude,
    required this.pentadLatitude,
    required this.province,
    required this.totalCards,
  });

  factory Pentad.fromJson(Map<String, dynamic> json) {
    return Pentad(
      pentadAllocation: json['pentad_Allocation'],
      pentadLongitude: json['pentad_Longitude']?.toDouble() ?? 0.0,
      pentadLatitude: json['pentad_Latitude']?.toDouble() ?? 0.0,
      province: Province.fromJson(json['province']),
      totalCards: json['total_Cards'],
    );
  }

  final String pentadAllocation;
  final double pentadLongitude;
  final double pentadLatitude;
  final Province province;
  final int totalCards;

  Map<String, Object?> toMap() {
    return {
      'pentadAllocation': pentadAllocation,
      'pentadLongitude': pentadLongitude,
      'pentadLatitude': pentadLatitude,
      'province': province.toMap(),
      'totalCards': totalCards,
    };
  }
}
